Filling TOF geo class

noferini opened the pull request:
Several items implemented:
- all TOF params only in one place (Geo class)
- fixes in the names of constants
- methods in Geo class optimized for pos-indexes conversion
- default rotation parameters computed once (init method in Geo class)

alibuild commented:
Error while checking build/o2/macos for 6d4d9dd85619c5567f2e8846ddf6b40560331dfe:
```

```
Full log [here](https://ali-ci.cern.ch/repo/logs/AliceO2Group/AliceO2/508/6d4d9dd85619c5567f2e8846ddf6b40560331dfe/fullLog.txt).

noferini commented:
Dear all,
I'm not able to reproduce the error (is it related to Mac OS?)
I need suppor for this.
Francesco

sawenzel commented:
@noferini : Don't worry about the macos setup. This is work in progress.

sawenzel commented on Detectors/TOF/base/include/TOFBase/Geo.h:
```diff
@@ -184,7 +196,35 @@ class Geo
+  static Int_t GetSector(const Float_t* pos);
+  static Int_t GetPlate(const Float_t* pos);
+  static Int_t GetPadZ(const Float_t* pos);
+  static Int_t GetPadX(const Float_t* pos);
```
in principle our coding guidelines ask for lower case letters for function names (which will be enforced at some moment); the start of the function should ideally be an action (verb)

sawenzel commented on Detectors/TOF/base/include/TOFBase/Geo.h:
```diff
@@ -184,7 +196,35 @@ class Geo
+  static Int_t GetPadZ(const Float_t* pos);
+  static Int_t GetPadX(const Float_t* pos);
+
+  static void FromGlobalToPlate(Float_t* pos, Int_t isector); // change coords to Plate reference
```
Is this modifying pos? the comment seems to refer to plate rather than sector (maybe sector should be called plate);

Would it make sense not to modify the original argument but to return the modified one? 
If these things are 3D cartesian coordinates, we should go towards the use of our Point3D class.

sawenzel commented on Detectors/TOF/base/include/TOFBase/Geo.h:
```diff
@@ -23,6 +23,15 @@ class Geo
 {
  public:
   // From AliTOFGeometry
+  static void Translation(Float_t* xyz, Float_t translationVector[3]);
```
consider just an action: `translate`

noferini commented:
I added one commit to fix the names.
In the meanwhile I fixed also some inconsistency in the TOF geometry.
I added a GetPos method which take the position of each channel from a geometry.root file.

Then I added a macro to verify that the mapping in Geo class is consistent with the geometry in Detector class (read from geometry.root). All the 160k channel indexes correspond in both classes (in future I will transfor this macro in a unit test)

noferini commented:
I forgot to answer this question:
>   private:
+  static void Init();
+
+  static Int_t GetSector(const Float_t* pos);
+  static Int_t GetPlate(const Float_t* pos);
+  static Int_t GetPadZ(const Float_t* pos);
+  static Int_t GetPadX(const Float_t* pos);
+
+  static void FromGlobalToPlate(Float_t* pos, Int_t isector); // change coords to Plate reference

Is this modifying pos? the comment seems to refer to plate rather than sector (maybe sector should be called plate);

Would it make sense not to modify the original argument but to return the modified one?
It is true that the method are modifying the position. This is what I want because that methods are private and used only in the GetDetId which get pos[3] as input but then work with a copy posLocal[3] which is updated at any step

alibuild commented:
Error while checking build/o2/macos for 54b8181ead87a661ba4cff9f34894dd99ad45967:
```
sw/BUILD/ROOT-latest/log
         ^
[ 73%] Building CXX object math/genvector/CMakeFiles/GenVector.dir/src/RotationZYX.cxx.o
[ 73%] Building CXX object net/net/CMakeFiles/Net.dir/src/TSecContext.cxx.o
1 error generated.
make[2]: *** [net/net/CMakeFiles/Net.dir/src/TSSLSocket.cxx.o] Error 1
make[2]: *** Waiting for unfinished jobs....
[ 73%] Building CXX object math/genvector/CMakeFiles/GenVector.dir/src/VectorUtil.cxx.o
[ 73%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/GenAlgoOptions.cxx.o
[ 74%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/GoFTest.cxx.o
[ 74%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/Integrator.cxx.o
[ 74%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/IntegratorOptions.cxx.o
[ 74%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/MersenneTwisterEngine.cxx.o
[ 74%] Linking CXX shared library ../../lib/libGenVector.so
make[1]: *** [net/net/CMakeFiles/Net.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 74%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/MinimizerOptions.cxx.o
[ 74%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/MinimTransformFunction.cxx.o
[ 74%] Built target GenVector
[ 74%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/MinimizerVariableTransformation.cxx.o
[ 74%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/MixMaxEngineImpl17.cxx.o
[ 74%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/MixMaxEngineImpl240.cxx.o
[ 74%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/MixMaxEngineImpl256.cxx.o
[ 74%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/ParameterSettings.cxx.o
[ 74%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/PdfFuncMathCore.cxx.o
[ 74%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/ProbFuncMathCore.cxx.o
[ 74%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/QuantFuncMathCore.cxx.o
[ 74%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/RandomFunctions.cxx.o
[ 74%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/RichardsonDerivator.cxx.o
[ 74%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/RootFinder.cxx.o
[ 74%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/SparseData.cxx.o
[ 74%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/SpecFuncCephes.cxx.o
[ 74%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/SpecFuncCephesInv.cxx.o
[ 74%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/SpecFuncMathCore.cxx.o
[ 74%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/StdEngine.cxx.o
[ 74%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/TComplex.cxx.o
[ 74%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/TKDTree.cxx.o
[ 74%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/TKDTreeBinning.cxx.o
[ 74%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/TMath.cxx.o
[ 74%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/TRandom.cxx.o
[ 74%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/TRandom1.cxx.o
[ 74%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/TRandom2.cxx.o
[ 74%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/TRandom3.cxx.o
[ 74%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/TRandomGen.cxx.o
[ 74%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/TStatistic.cxx.o
[ 74%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/UnBinData.cxx.o
[ 74%] Building C object math/mathcore/CMakeFiles/MathCore.dir/src/triangle.c.o
[ 74%] Linking CXX shared library ../../lib/libMathCore.so
[ 74%] Built target MathCore
[ 74%] Built target onepcm
```
Full log [here](https://ali-ci.cern.ch/repo/logs/AliceO2Group/AliceO2/508/54b8181ead87a661ba4cff9f34894dd99ad45967/fullLog.txt).

alibuild commented:
Error while checking build/O2/o2-dev-fairroot for 54b8181ead87a661ba4cff9f34894dd99ad45967:
```

```
Full log [here](https://ali-ci.cern.ch/repo/logs/AliceO2Group/AliceO2/508/54b8181ead87a661ba4cff9f34894dd99ad45967/fullLog.txt).

alibuild commented:
Error while checking build/o2/macos for 8b129e303bd2d976d232a05b0d217afbb12a22f9:
```
sw/BUILD/FairRoot-latest/log
CMake Error at fairmq/CMakeLists.txt:246 (add_executable):
  Target "shmmonitor" links to target "Boost::log_setup" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?


CMake Error at fairmq/CMakeLists.txt:246 (add_executable):
  Target "shmmonitor" links to target "Boost::program_options" but the target
  was not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?


CMake Error at fairmq/CMakeLists.txt:246 (add_executable):
  Target "shmmonitor" links to target "Boost::thread" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?


CMake Error at fairmq/CMakeLists.txt:246 (add_executable):
  Target "shmmonitor" links to target "Boost::system" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?


CMake Error at fairmq/CMakeLists.txt:246 (add_executable):
  Target "shmmonitor" links to target "Boost::filesystem" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?


CMake Error at fairmq/CMakeLists.txt:246 (add_executable):
  Target "shmmonitor" links to target "Boost::regex" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?


CMake Error at fairmq/CMakeLists.txt:246 (add_executable):
  Target "shmmonitor" links to target "Boost::date_time" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?


-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    MACOSX_RPATH


```
Full log [here](https://ali-ci.cern.ch/repo/logs/AliceO2Group/AliceO2/508/8b129e303bd2d976d232a05b0d217afbb12a22f9/fullLog.txt).
